core/services/ocr2/plugins/ccip/internal/pricegetter/evm.go

```diff
@@ -77,15 +79,20 @@ func NewDynamicPriceGetter(cfg config.DynamicPriceGetterConfig, evmClients map[u
 
 // TokenPricesUSD implements the PriceGetter interface.
 // It returns static prices stored in the price getter, and batch calls to aggregators (on per chain) for aggregator-based prices.
-func (d *DynamicPriceGetter) TokenPricesUSD(ctx context.Context, tokens []common.Address) (map[common.Address]*big.Int, error) {
-	prices := make(map[common.Address]*big.Int, len(tokens))
+func (d *DynamicPriceGetter) TokenPricesUSD(ctx context.Context, tokens []cciptypes.Address) (map[cciptypes.Address]*big.Int, error) {
+	prices := make(map[cciptypes.Address]*big.Int, len(tokens))
 
 	batchCallsPerChain := make(map[uint64][]rpclib.EvmCall)
 
 	// required to maintain the order of the batched rpc calls for mapping the results
 	batchCallsTokensOrder := make(map[uint64][]common.Address)
 
-	for _, tk := range tokens {
+	evmAddrs, err := ccipcalc.GenericAddrsToEvm(tokens...)
+	if err != nil {
+		return nil, fmt.Errorf("addresses are non-evm: %w", err)
```

Comment:
Let’s return the original error here. Otherwise, the error message becomes repetitive (`GenericAddrsToEvm` already returns a similar error message).
```suggestion
		return nil, err
```

Reply:
nice catch